src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,154 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
   }
-
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
     log.info("Reregistered framework.");
   }
 
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeTask(status.getTaskId());
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status); // TODO(rubbish) could terminal/running get pushed to
+                                             // this
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_1);
+          }
+          break;
+        case NAME_NODE_1 :
+          if (liveState.getNameNodeSize() == 1 && liveState.getFirstNameNodeTaskId() != null) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_2);
+          } else {
+            log.info("Cannot locate first namenode task id");
+          }
+          break;
+        case NAME_NODE_2 :
+          if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES
+              && liveState.getSecondNameNodeTaskId() != null) {
+            reloadConfigsOnAllRunningTasks(driver);
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(), liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_INIT_MESSAGE);
+            blockUntilNameNode1Healthy();
```

Comment:
In the current code it doesn't send status of running until the NN is actually launched. I'm curious aboutwhat changed here. Would it be possible to investigate this, because the sleep is quite ugly. We could always put it in temporarily as long as a subsequent PR was done to fix it, but I'd prefer not to.